contracts/helpers/TransferHelper.sol

```diff
@@ -131,6 +202,21 @@ contract TransferHelper is TransferHelperInterface, TokenTransferrer {
                 )
             );
 
+            // Create a variable to store the codehash of the conduit.
+            bytes32 codeHash;
+
+            // Retrieve the codehash of the conduit and assign it to codeHash.
+            assembly {
+                codeHash := extcodehash(conduit)
```

Comment:
This step is not necessary — instead, we should ensure that the correct magic value is returned when calling the conduit

Reply:
if there is no contract at the address (i.e. a conduitKey was supplied that doesn't exist) then it won't return a magic value. In fact, it won't do anything at all, except revert with no reason string, with no way we've been able to capture it.

@stephankmin and I found this strange and a similar question [here](https://ethereum.stackexchange.com/questions/128845/why-does-soliditys-try-catch-not-catch-call-to-non-contract-address-revertion) with no answer.

so extcodehash was the cheapest way we're able to identify if it is indeed first a contract account, which uses [400 gas](https://eips.ethereum.org/EIPS/eip-1052#specification). then we can compare the magic value to separate non-conforming conduits.

Reply:
if there is really no way to capture it (we should keep looking into it) and we think it's worth saving the 400 gas check per transfer then it could just be the defined behavior that would have failed anyway, it just has no message.